### PR TITLE
fix: prioritization of icon rule

### PR DESCRIPTION
**Pull request by Mara-Li**
See #66 
I added a new function "check if folder has an icon" used during custom rule : It will prevent to adding icon if specific settings are found.

### CODE REVIEW

**Comment by FlorianWoelki** [comment]
Does it fix the issue #66 ?

File: src/util.ts [hunk 1]
```diff
@@ -198,15 +197,18 @@ export const addIconsToDOM = (
         const regex = new RegExp(rule.rule);
         plugin.app.vault.getAllLoadedFiles().forEach(async (file) => {
           const fileType = (await plugin.app.vault.adapter.stat(file.path)).type;
-          if (file.name.match(regex) && isToRuleApplicable(rule, fileType)) {
+          const settings_folder = checkIfFolderHasIconsSettings(plugin, file.path);
```
Comment (FlorianWoelki) [inline src/util.ts:200]: Please rename the variable to camelCase.

File: src/util.ts [hunk 2]
```diff
@@ -198,15 +197,18 @@ export const addIconsToDOM = (
...
+          if (file.name.match(regex) && isToRuleApplicable(rule, fileType) && !settings_folder) {
             addCustomRuleIcon(rule, file.path);
           }
         });
       } catch {
         // Rule is not applicable to a regex format.
+
         plugin.app.vault.getAllLoadedFiles().forEach(async (file) => {
+          const settings_folder = checkIfFolderHasIconsSettings(plugin, file.path);
```
Comment (FlorianWoelki) [inline src/util.ts:209]: Here as well.

**Comment by Mara-Li** [comment]
Oups, Python habit xD !
Done.
I also edit the check to check the inheritance. For me, the order must be :
custom rule > inheritance > specific customization

If you don't like that (maybe it was intended), maybe add a simple checker in settings?

**Pull request merged**